src/modules/launcher/Plugins/Microsoft.Plugin.Program/Main.cs

```diff
@@ -86,8 +86,11 @@ public List<Result> Query(Query query)
                 .Where(p => p.Enabled)
                 .Select(p => p.Result(query.Search, _context.API));
 
-            var result = results1.Concat(results2).Where(r => r != null && r.Score > 0).ToList();
-            return result;
+            var result = results1.Concat(results2).Where(r => r != null && r.Score > 0);
+            var maxScore = result.Max(x => x.Score);
+            result = result.Where(x => x.Score > _settings.MinScoreThreshold * maxScore);

```

Comment:
Just thinking out loud, do we want to use the threshold approach or do we want to cap the number of results to a fixed value?

Reply:
The number of results with the same high score should be less. For example, querying "Ma" would have a high score for `Mail`, `Maps`, and `Magnify` but there can be a number of programs with `Ma` as a substring in the exe name and low score. Capping a number of programs can end up including results with low scores as well even though they are not very relevant results.